src/PythNetworkHTTPClient.ts

```diff
@@ -0,0 +1,153 @@
+        this._program_key = getPythProgramKeyForCluster(cluster_name)        
+        this._commitment = commitment;
+        this._cluster = cluster_name;
+        this._price_queue = [];
+        this.productAccountKeyToProduct = {};
+        this._asset_types = new Set();
+        this._product_symbols = new Set();
+        this._products = new Set();
+        this._product_price = new Map<Product, PriceData>();
+        this.priceAccountKeyToProductAccountKey = {}
+
+    }
+
+    private handleProductAccount(key: PublicKey, account: AccountInfo<Buffer>) {
+        const {priceAccountKey, type, product} = parseProductData(account.data)
+        this.productAccountKeyToProduct[key.toString()] = product
+        if (priceAccountKey.toString() !== ONES) {
+            this.priceAccountKeyToProductAccountKey[priceAccountKey.toString()] = key.toString()
+        }
+
+        this._asset_types.add(product.asset_type);
+        this._product_symbols.add(product.symbol);
+        this._products.add(product);
+    }
+
+    private handlePriceAccount(key: PublicKey, account: AccountInfo<Buffer>) {
+        const product = this.productAccountKeyToProduct[this.priceAccountKeyToProductAccountKey[key.toString()]]
+        if (product === undefined) {
+            // This shouldn't happen since we're subscribed to all of the program's accounts,
+            // but let's be good defensive programmers.
+            throw new Error('Got a price update for an unknown product. This is a bug in the library, please report it to the developers.')
+        }
+
+        const priceData = parsePriceData(account.data)     
+        this._product_price.set(product, priceData);
+    }    
+
+    private handleAccount(key: PublicKey, account: AccountInfo<Buffer>, productOnly: boolean) {
+        const base = parseBaseData(account.data)
+        // The pyth program owns accounts that don't contain pyth data, which we can safely ignore.
+        if (base) {
+            switch (AccountType[base.type]) {
+                case 'Mapping':
+                    // We can skip these because we're going to get every account owned by this program anyway.
+                    break;
+                case 'Product':
+                    this.handleProductAccount(key, account)
+                    break;
+                case 'Price':
+                    if (!productOnly) {
+                        this.handlePriceAccount(key, account)
+                    } else {
+                        this._price_queue.push({
+                            key: key,
+                            account: account
+                        });
+                    }
+                    break;
+                case 'Test':
+                    break;
+                default:
+                    throw new Error(`Unknown account type: ${base.type}. Try upgrading pyth-client.`)
+            }
+        }
+    }
+
+    public async refreshData() {
```

Comment:
I recommend renaming this method something like `getData()` and changing the signature to return an object containing the price information for all products. That approach avoids mutable state inside the client object, which should generally make this easier to use and reduce bugs for downstream users.

In general, the mutable state on this object looks like it's unnecessary. Instead, it could all be local variables inside this method.

Reply:
also, can you leave a doc comment on every public method explaining what it does?